--- busybox-1.9.0.orig/networking/wget.c	2008-01-04 08:11:50.000000000 +0000
+++ busybox-1.9.0/networking/wget.c	2008-01-04 08:52:20.000000000 +0000
@@ -33,6 +33,8 @@
 	unsigned start_sec;
 #endif
 	smallint chunked;             /* chunked transfer encoding */
+	smallint started;
+	smallint print_progress;
 };
 #define G (*(struct globals*)&bb_common_bufsiz1)
 struct BUG_G_too_big {
@@ -47,14 +49,23 @@
 #define lastupdate_sec  (G.lastupdate_sec )
 #define start_sec       (G.start_sec      )
 #define chunked         (G.chunked        )
+#define started         (G.started        )
+#define print_progress  (G.print_progress )
 #define INIT_G() do { } while (0)
 
-
 #if ENABLE_FEATURE_WGET_STATUSBAR
 enum {
 	STALLTIME = 5                   /* Seconds when xfer considered "stalled" */
 };
 
+static void timeout_reached(int flag)
+{
+	if(started == 1)
+		return;
+	fprintf(stderr, "Timeout.\n");
+	exit(1);
+}
+
 static int getttywidth(void)
 {
 	int width;
@@ -85,6 +96,7 @@
 		if (ratio > 100) ratio = 100;
 	}
 
+	if (print_progress == FALSE) {
 	fprintf(stderr, "\r%-20.20s%4d%% ", curfile, ratio);
 
 	barlength = getttywidth() - 49;
@@ -137,6 +149,9 @@
 			fprintf(stderr, "%02d:%02d:%02d ETA", eta / 3600, i / 60, i % 60);
 		}
 	}
+	} else {
+		fprintf(stderr, "%3d%%\n", ratio);
+	}
 
 	if (flag == 0) {
 		/* last call to progressmeter */
@@ -404,6 +419,7 @@
 	char *str;
 	char *proxy = 0;
 	char *dir_prefix = NULL;
+	char *timeout = NULL;
 #if ENABLE_FEATURE_WGET_LONG_OPTIONS
 	char *extra_headers = NULL;
 	llist_t *headers_llist = NULL;
@@ -430,8 +446,10 @@
 		WGET_OPT_PREFIX     = 0x10,
 		WGET_OPT_PROXY      = 0x20,
 		WGET_OPT_USER_AGENT = 0x40,
-		WGET_OPT_PASSIVE    = 0x80,
-		WGET_OPT_HEADER     = 0x100,
+		WGET_OPT_PRINT_P    = 0x80,
+		WGET_OPT_TIMEOUT    = 0x100,
+		WGET_OPT_PASSIVE    = 0x200,
+		WGET_OPT_HEADER     = 0x400,
 	};
 #if ENABLE_FEATURE_WGET_LONG_OPTIONS
 	static const char wget_longopts[] ALIGN1 =
@@ -443,6 +461,8 @@
 		"directory-prefix\0" Required_argument "P"
 		"proxy\0"            Required_argument "Y"
 		"user-agent\0"       Required_argument "U"
+		"print-progress\0"   No_argument       "g"
+		"timeout\0"          Required_argument "T"
 		"passive-ftp\0"      No_argument       "\xff"
 		"header\0"           Required_argument "\xfe"
 		;
@@ -455,15 +475,16 @@
 #endif
 	/* server.allocated = target.allocated = NULL; */
 	opt_complementary = "-1" USE_FEATURE_WGET_LONG_OPTIONS(":\xfe::");
-	opt = getopt32(argv, "csqO:P:Y:U:",
+	opt = getopt32(argv, "csqO:P:Y:U:gT:",
 				&fname_out, &dir_prefix,
-				&proxy_flag, &user_agent
+				&proxy_flag, &user_agent, &timeout
 				USE_FEATURE_WGET_LONG_OPTIONS(, &headers_llist)
 				);
 	if (strcmp(proxy_flag, "off") == 0) {
 		/* Use the proxy if necessary */
 		use_proxy = 0;
 	}
+
 #if ENABLE_FEATURE_WGET_LONG_OPTIONS
 	if (headers_llist) {
 		int size = 1;
@@ -515,6 +536,19 @@
 	curfile = bb_get_last_path_component_nostrip(fname_out);
 #endif
 
+	print_progress = FALSE;
+	if (opt & WGET_OPT_PRINT_P)
+		print_progress = TRUE;
+
+	if (opt & WGET_OPT_TIMEOUT) {
+		int timeout_sec;
+		if(sscanf(timeout, "%d", &timeout_sec) == 1)
+		{
+			signal(SIGALRM, timeout_reached);
+			alarm(timeout_sec);
+		}
+	}
+
 	/* Impossible?
 	if ((opt & WGET_OPT_CONTINUE) && !fname_out)
 		bb_error_msg_and_die("cannot specify continue (-c) without a filename (-O)"); */
@@ -745,6 +779,8 @@
 	 * Retrieve file
 	 */
 
+	started = 1;
+
 	/* Do it before progressmeter (want to have nice error message) */
 	if (output_fd < 0) {
 		int o_flags = O_WRONLY | O_CREAT | O_TRUNC | O_EXCL;
