--- busybox-1.1.3.p2/util-linux/mdev.c	2006-03-22 16:16:26 -0500
+++ busybox-1.1.3/util-linux/mdev.c	2006-06-13 08:30:42 -0400
@@ -200,6 +200,10 @@
 		if (entry->d_name[0] == '.')
 			continue;
 
+		if ((!strcmp(entry->d_name, "device")) || (!strcmp(entry->d_name, "bridge")))
+			/* It's a link */
+			continue;
+
 		if (entry->d_type == DT_DIR) {
 			snprintf(path+len, PATH_MAX-len, "/%s", entry->d_name);
 			find_dev(path);
